packages/react-ui/src/app/features/ai/lib/use-ai-assistant-chat-history.ts

```diff
@@ -0,0 +1,59 @@
+import { toast } from '@openops/components/ui';
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import { t } from 'i18next';
+import { aiAssistantChatHistoryApi } from './ai-assistant-chat-history-api';
+
+export function useAssistantChatHistory() {
+  const qc = useQueryClient();
+
+  const { data, isLoading } = useQuery({
+    queryKey: ['assistant-history'],
+    queryFn: async () => {
+      const res = await aiAssistantChatHistoryApi.list();
+      return res.chats ?? [];
+    },
+    select: (chats) =>
+      chats.map((c) => ({
+        id: c.chatId,
+        displayName: c.chatName || 'New chat',
+      })),
+    refetchOnWindowFocus: false,
```

Comment:
could have new chats in other tabs, I think the refetch on window focus makes sense